File: maximum-length-of-repeated-subarray/maximum-length-of-repeated-subarray.cpp
class Solution {
public:
    int findLength(vector<int>& nums1, vector<int>& nums2) {
        int n=nums1.size(),m=nums2.size();
        vector<vector<int>>v(n+1,vector<int>(m+1,0));
        for(int i=1;i<=n;i++){
            for(int j=1;j<=m;j++){
                if(nums1[i-1]==nums2[j-1])
                    v[i][j]=v[i-1][j-1]+1;
                else
                    v[i][j]=0;
            }
        }
        int ans=0;
        for(auto x:v){
            for(auto y:x){
                ans=max(y,ans);
            }
        }
        return ans;
    }
};

File: 718-maximum-length-of-repeated-subarray/maximum-length-of-repeated-subarray.py
class Solution:
    def findLength(self, nums1: List[int], nums2: List[int]) -> int:
        def solve(nums1, nums2):
            n = len(nums1)
            m = len(nums2)
            ans = 0
            
            # Try all possible alignments of nums1 and nums2
            for i in range(n):
                l = 0
                for j in range(min(m, n - i)):
                    if nums1[i + j] == nums2[j]:
                        l += 1
                        ans = max(ans, l)
                    else:
                        l = 0
            
            return ans
        
        # Check both alignments
        return max(solve(nums1, nums2), solve(nums2, nums1))
